### DISCUSSION TITLE
Moving shards

### DISCUSSION THREAD

**Discussion by jrmy911** [apache/couchdb#4544] [Q&A]
I found the documentation about [moving shards](https://docs.couchdb.org/en/stable/cluster/sharding.html) and got through the first steps by enabling maintenance mode. 

I haven't got a ton of experience with JSON but I was wondering how I could update the changelog, by_node and by_range. I feel like the documentation "skips" the CLI for this:
`To add a shard to a node, add entries like this to the database metadata’s changelog attribute:
["add", "{range}", "{node-name}"]`
It goes rather quick on how about to add that fourth node, if using the example in the documentation.

Okay, but how does this translate into a CLI command using `curl -X PUT -H "Content-type: application/json"` ? That's where my knowledge is limited. I tried different things like:
`curl -X PUT -H "Content-type: application/json" http://{hostname}/_node/_local/_dbs/{dbname} -d '["add","{range}","{node-name}"]'` 
`curl -X PUT -H "Content-type: application/json" http://{hostname}/_node/_local/_dbs/{dbname} -d '["add",{range},{node-name}]'` 

But it eventually fails and it is either invalid JSON or `{"error":"bad_request","reason":"Document must be a JSON object"}`.
I'm sure I'm not entirely using the correct syntax, but after having gone through CouchDB's documentation (and the JSON API page), I am feeling a bit lost. :)

**Comment by big-r81** [comment]
Hi,

you need to: 

1. download the database metadata document
2. add the following entries to that metadata document
    * add appropriate `changelog` entries
    * update the `by_node` entries
    * update the `by_range` entries
3. Upload the (edited) document back to CouchDB 
    `$ curl -X PUT http://localhost/_node/_local/_dbs/{name} -d '{...}'`

This is an advanced task, sometime it's easier to set up a new cluster and replicate the old data to the new cluster.
